src/extensions/default/InAppNotifications/main.js

```diff
@@ -0,0 +1,372 @@
+                    PreferencesManager.setViewState("notificationInfo", notificationInfo);
+                    result.resolve(notificationInfo);
+                }).fail(function (jqXHR, status, error) {
+                    // When loading data for unit tests, the error handler is
+                    // called but the responseText is valid. Try to use it here,
+                    // but *don't* save the results in prefs.
+
+                    if (!jqXHR.responseText) {
+                        // Text is NULL or empty string, reject().
+                        result.reject();
+                        return;
+                    }
+
+                    try {
+                        data = JSON.parse(jqXHR.responseText);
+                        result.resolve(data);
+                    } catch (e) {
+                        result.reject();
+                    }
+                });
+            });
+        } else {
+            result.resolve(data);
+        }
+
+        return result.promise();
+    }
+
+
+    /**
+     * Check for notifications, notification overlays are always displayed
+     *
+     * @return {$.Promise} jQuery Promise object that is resolved or rejected after the notification check is complete.
+     */
+    function checkForNotification() {
+        var result = new $.Deferred();
+        var versionInfoUrl;
+
+        _getNotificationInformation(versionInfoUrl)
+            .done(function (notificationInfo) {
+                // Get all available notifications
+                var notifications = notificationInfo.notifications;
+                if (notifications && notifications.length > 0) {
+                    // Iterate through notifications and act only on the most recent
+                    // applicable notification
+                    notifications.every(function(notificationObj) {
+                        // Only show the notification overlay if the user hasn't been
+                        // alerted of this notification
+                        if (_checkNotificationValidity(notificationObj)) {
+                            if (notificationObj.silent) {
+                                // silent notifications, to gather user validity based on filters
+                                HealthLogger.sendAnalyticsData("notification", notificationObj.sequence, "handled");
+                            } else {
+                                showNotification(notificationObj);
+                            }
+                            // Break, we have acted on one notification already
+                            return false;
+                        }
+                    });
+                }
+                result.resolve();
+            })
+            .fail(function () {
+                // Error fetching the update data. If this is a forced check, alert the user
+                result.reject();
+            });
+
+        return result.promise();
```

Comment:
nit: I think we can directly return `_getNotificationInformation()` promise instead of creating a newer promise.

Reply:
Need to add a parameter in the wrapper function which can be provided for testing.